### PR TITLE
feat(checkbox): #106: Add checkbox component

### PR DESCRIPTION
**Pull request by jhsu**
Add a checkbox component based on radix-ui

image

image

### CODE REVIEW

**Comment by jhsu** [comment]
cc @dugdaniels @dvargas92495 
(I don't have access to add reviewers)

File: packages/components/components/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  display: "inline-flex",
+  alignItems: "center",
+  justifyContent: "center",
+  defaultVariants: {
+    checkDisplay: "unchecked",
+  },
+  variants: {
+    disabledClick: {
+      true: {
+        backgroundColor: "#FFEED9",
```
Comment (jhsu) [inline packages/components/components/Checkbox/Checkbox.tsx:22]: we use this in a few places, we likely will want to pull this to a theme variable, probably separated.

Comment (dvargas92495) [reply packages/components/components/Checkbox/Checkbox.tsx:22]: yea we should add it to here, could be a separate PR

https://github.com/CabinDAO/Topo/blob/main/packages/theme/src/index.ts#L23-L34

File: packages/components/components/Checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+export interface CheckboxProps
+  extends Omit<RadixCheckbox.CheckboxProps, "css"> {}
+const Checkbox = (props: CheckboxProps) => {
+  const [checked, setChecked] = useState(props.checked ?? false);
+  return (
+    <StyledBox
+      {...props}
+      disabledClick={props.disabled}
+      checkDisplay={checked ? "checked" : "unchecked"}
```
Comment (dvargas92495) [inline packages/components/components/Checkbox/Checkbox.tsx:53]: why not just use the boolean?

**Review by dvargas92495 — APPROVED** [review]
🎉 Nice! left a question, feel free to merge anyway

File: packages/components/components/Checkbox/Checkbox.tsx (hunk 2 above)
Comment (jhsu) [reply packages/components/components/Checkbox/Checkbox.tsx]: discovered boolean variants after i did this 😄 

I also greatly simplified this to not require variants and instead use attribute selectors (`data-state`) provided by radix-ui

**Comment by jhsu** [comment]
@dvargas92495 fixed this up a bunch, i don't have permissions to merge this

File: packages/components/components/Checkbox/Checkbox.tsx (hunk 2 above)
Comment (dvargas92495) [reply packages/components/components/Checkbox/Checkbox.tsx]: hell yea 🔥

**Pull request merged**